sklearn/ensemble/weight_boosting.py

```diff
@@ -234,10 +239,15 @@ def staged_score(self, X, y, sample_weight=None):
             else:
                 yield r2_score(y, y_pred, sample_weight=sample_weight)
 
-    @property
-    def feature_importances_(self):
-        """Return the feature importances (the higher, the more important the
-           feature).
+    def get_feature_importances_(self, norm_order=2):
```

Comment:
It's okay. I will come-up with a solution by Monday.

Reply:
@jnothman: How about we make a Mixin for `feature_importances_` and inherit it where-ever it's required (Adaboost estimator, RFE, SelectFromModel etc..). This will require `norm_order` to be declared as a class variable in which-ever class it is being inherited into.

The job of this mixin is to determine whether a metaestimator or an usual estimator is inheriting it and then calculate `feature_importances_` accordingly. 

Something like this that can go into sklearn/base.py: 

```
class FeatureImportancesMixin(object):
    """Mixin class that generates ``feature_importances_`` attribute."""

    def _is_metaestimator(self):
        """Determines if `FeatureImportancesMixin` is inherited by an
        metaestimator.
        """
        return hasattr(self, "estimators_")

    def _get_feature_importances(estimator, norm_order):
        """Retrieve or aggregate feature importances from estimator.

        Parameters
        ----------
        estimator : any estimator object with either `feature_importances_`
            attribute or `coef_` attribute

        norm_order : {non-zero int, inf, -inf, "fro"}, optional
            Same as `numpy.linalg.norm`'s norm_order parameter

        Returns
        -------
        importances : A 1d array, shape: [n_features]
        """
        importances = getattr(estimator, "feature_importances_", None)

        coef_ = getattr(estimator, "coef_", None)
        if importances is None and coef_ is not None:
            if estimator.coef_.ndim == 1:
                importances = np.abs(coef_)

            else:
                importances = np.linalg.norm(coef_,
                                             axis=0,
                                             ord=self.norm_order)

        elif importances is None:
            raise AttributeError(
                "The underlying estimator %s has no `coef_` or "
                "`feature_importances_` attribute. Either pass a fitted estimator"
                " to SelectFromModel or call fit before calling transform."
                % estimator.__class__.__name__)

        return importances

    @property
    def feature_importances_(self):
        """Return the feature importances.

         The higher the value, the more important the feature.

        Returns
        -------
        feature_importances_ : array, shape = [n_features]
        """

           try:

                if self._is_metaestimator():
                    if self.estimators_ is None or len(self.estimators_) == 0:
                        raise ValueError(
                        "Estimator not fitted, "
                        "call `fit` before `feature_importances_`.")

                    norm = self.estimator_weights_.sum()
                    estimator_feature_importances_ = [
                        _get_feature_importances(estimator, self.norm_order)
                        for estimator in self.estimators_]

                    return (sum(weight * feature_importances_
                                for weight, feature_importances_ in zip(
                                    self.estimator_weights_,
                                    estimator_feature_importances_)) / norm)
                else:
                    return _get_feature_importances(estimator=self.estimator_,
                                                    norm_order=self.norm_order)

            except AttributeError:
                raise AttributeError(
                    "Unable to compute feature importances "
                    "since base estimator neither has a "
                    "feature_importances_ attribute nor a "
                    "coef_ attribute")
```